tbc/sim.go

```diff
@@ -150,74 +150,96 @@ func (sim *Simulation) reset() {
 	sim.Agent.Reset(sim)
 }
 
+// Activates set bonuses, returning the list of active bonuses.
+func (sim *Simulation) ActivateSets() []string {
+	active := []string{}
+	// Activate Set Bonuses
+	for _, set := range sets {
+		itemCount := 0
+		for _, i := range sim.Equip {
+			if set.Items[i.Name] {
+				itemCount++
+				if bonus, ok := set.Bonuses[itemCount]; ok {
+					active = append(active, set.Name+" ("+strconv.Itoa(itemCount)+"pc)")
+					sim.addAura(bonus(sim))
+				}
+			}
+		}
+	}
+	return active
+}
+
 // Run will run the simulation for number of seconds.
 // Returns metrics for what was cast and how much damage was done.
-func (sim *Simulation) Run(seconds int) SimMetrics {
-	sim.endTick = seconds * TicksPerSecond
+func (sim *Simulation) Run(durationSeconds float64) SimMetrics {
+	sim.endTime = durationSeconds
 	sim.reset()
 
-	for sim.CurrentTick < sim.endTick {
-		sim.Spellcasting()
+	for sim.CurrentTime < sim.endTime {
+		TryActivateDrums(sim)
+		TryActivateBloodlust(sim)
+		TryActivateEleMastery(sim)
+		TryActivateRacial(sim)
+		TryActivateDestructionPotion(sim)
+		sim.TryActivateEquipment()
+
+		didPot := TryActivateDarkRune(sim) || TryActivateSuperManaPotion(sim)
```

Comment:
This will cause super mana pot to never be used if you used a dark rune. Not sure if it matters that much, but it means that you won't use the mana pot until after the next cast even if you could have used it now. 

Again, not sure if that matters in practical terms, but this change does change the behavior of the sim. We should figure out if we care.

Reply:
Ah good point, this was not intended. Yeah it shouldn't matter but I'll prefer to keep the same behavior for now. In another PR I'll be removed the didPot variable entirely so this won't be an issue.